src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -156,13 +157,17 @@ private Task<LocationOrLocationLinks> HandleObjectPropertyLocationAsync(Definiti
             if (matchingNodes[1] is ModuleDeclarationSyntax moduleDeclarationSyntax)
             {
                 // capture the property accesses leading to this specific property access
-                var propertyAccesses = matchingNodes.OfType<ObjectPropertySyntax>()
-                    .Select(node => node.TryGetKeyText())
-                    .OfType<string>().ToList();
-                // only two level of traversals: mod.outputs.<outputName> or mod.params.<parameterName>
+                var propertyAccesses = matchingNodes.OfType<ObjectPropertySyntax>().ToList();
+                // only two level of traversals: mod { params: { <outputName1>: ...}}
                 if (propertyAccesses.Count == 2)
                 {
-                    return GetModuleSymbolLocationAsync(result, context, moduleDeclarationSyntax, propertyAccesses[0], propertyAccesses[1]);
+                    // underline only the key of the object property access
+                    return GetModuleSymbolLocationAsync(
+                        propertyAccesses.Last().Key, 
+                        context, 
+                        moduleDeclarationSyntax,
+                        propertyAccesses[0].TryGetKeyText()!, 
+                        propertyAccesses[1].TryGetKeyText()!);
```

Comment:
Check for null rather than using `!` - using `!` means there's a risk this method could throw unexpectedly.

Reply:
Yeah we generally want to avoid null suppression on the product side. (Fair game in tests, though.)